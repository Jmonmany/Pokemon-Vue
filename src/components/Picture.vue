<template>
  <div class="pokemon-container">
    <img :src="imgSrc" alt="pokemon" class="hidden" />
    <img v-if="showPokemon" :src="imgSrc" alt="pokemon" class="fade-in" />
  </div>
</template>

<script>
export default {
  props: {
    pokemonId: {
      type: Number,
      required: true,
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
    },
  },
};
</script>

<style scoped>
img {
  height: 200px;
  position: absolute;
  right: 44%;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden {
  filter: brightness(0);
}
.pokemon-container {
  height: 200px;
}
</style>
