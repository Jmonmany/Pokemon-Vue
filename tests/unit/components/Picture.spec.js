import { shallowMount } from "@vue/test-utils";
import Picture from "@/components/Picture";

describe("Picture.vue component", () => {
  test("should match with the snapshot", () => {
    const wrapper = shallowMount(Picture, {
      props: {
        pokemonId: 1,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("should render the hidden img and the pokemon 100", () => {
    const wrapper = shallowMount(Picture, {
      props: {
        pokemonId: 100,
        // showPokemon: false, by default is false
      },
    });
    const url = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${wrapper.props().pokemonId}.svg`;
    const [img1, img2] = wrapper.findAll("img");
    expect(img1.exists()).toBeTruthy();
    expect(img2).toBe(undefined);
    expect(img1.classes("hidden")).toBeTruthy();
    expect(img1.attributes("src")).toBe(url);
  });
  test("should render the img when showPokemon is true", () => {
    const wrapper = shallowMount(Picture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    });
    const [img1, img2] = wrapper.findAll("img");
    expect(img2.exists()).toBeTruthy();
    expect(img1.classes("hidden")).toBeTruthy();
    expect(img2.classes("fade-in")).toBeTruthy();
  });
});
