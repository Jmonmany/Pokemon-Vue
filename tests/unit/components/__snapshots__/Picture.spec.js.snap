// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture.vue component should match with the snapshot 1`] = `
<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon" class="hidden">
  <!--v-if-->
</div>
`;
